core-ui/src/components/Extensibility/helpers/index.js

```diff
@@ -89,6 +90,41 @@ export function createTemplate(api, namespace, scope) {
   return template;
 }
 
+export function getDefaultPreset(presets, emptyTemplate) {
+  if (!presets || !presets.length) return null;
+  const defaultPreset = presets.find(preset => preset.default === true);
+  return defaultPreset
+    ? merge({}, { value: emptyTemplate }, defaultPreset)
+    : null;
+}
+
+export function PreparePresets(resource, presets) {
```

Comment:
Could you please change it to a custom hook instead of a React component? React components starts with a capital letter but they should return a JSX. It doesn't return a JSX but an array of prepared presets, so it should be changed to a custom hook.